doc/shield-coinbase.md

```diff
@@ -0,0 +1,101 @@
+
+## Background
+
+The current Zcash protocol includes a consensus rule that coinbase rewards must be sent to a shielded address.
+
+## User Experience Challenges
+
+A user can use the z_sendmany RPC call to shield coinbase funds, but the call was not designed for sweeping up many utxos, and offered a suboptimal user experience.
+
+If customers send mining pool payouts to their online wallet, the service provider must sort through UTXOs to to correctly determine the non-coinbase UTXO funds that can be withdrawn or transferred by customers to another transparent address.
+
+## Solution
+
+The z_shieldcoinbase call makes it easy to sweep up coinbase rewards from multiple coinbase utxos across multiple coinbase reward addresses.
+
+    z_shieldcoinbase fromaddress toaddress (fee) (limit)
+
+Where the default fee is 0.0010000 ZEC and the limit is 50.
+
+## Examples
+
+Sweep up coinbase utxos from a transparent address you use for mining:
+
+    zcash-cli z_shieldcoinbase tMyMiningAddress zMyPrivateAddress
+
+Sweep up coinbase utxos from multiple transparent addresses to a shielded address:
+
+    zcash-cli z_shieldcoinbase "*" zMyPrivateAddress
+
+Sweep up with a fee of 1.23 ZEC:
+
+    zcash-cli z_shieldcoinbase tMyMiningAddress zMyPrivateAddress 1.23
+
+Sweep up with a fee of 0.1 ZEC and set limit on the maximum number of utxos to shield at 25:
+
+    zcash-cli z_shieldcoinbase "*" zMyPrivateAddress 0.1 25
+
+### Asynchronous Call
+
+The `z_shieldcoinbase` RPC call is an asynchronous call, so you can queue up multiple operations. 
+
+When you invoke
+
+    zcash-cli z_shieldcoinbase tMyMiningAddress zMyPrivateAddress
+
+JSON will be returned immediately, with the following data fields populated:
+
+- operationid: a temporary id to use with `z_getoperationstatus` and `z_getoperationresult` to get the status and result of the operation.
+- shieldedUTXOs: number of coinbase utxos being shielded
+- shieldedValue: value of coinbase utxos being shielded.
+- remainingUTXOs: number of coinbase utxos still available for shielding.
+- remainingValue: value of coinbase utxos still available for shielding
+
+### Locking UTXOs
+
+The `z_shieldcoinbase` call will lock any selected UTXOs. This prevents the selected UTXOs which are already queued up from being selected for any other send operation.  If the `z_shieldcoinbase` call fails, any locked UTXOs are unlocked.
+
+You can use the RPC call `lockunspent` to see which UTXOs have been locked.  You can also use this call to unlock any UTXOs in the event of an unexpected system failure which leaves UTXOs in a locked state.
+
+### Limits, Performance and Transaction Confirmation
+
+The number of coinbase utxos selected for shielding can be adjusted by setting the limit parameter. The default value is 50.
+
+If the limit parameter is set to zero, the zcashd `mempooltxinputlimit` option will be used instead, where the default value for `mempooltxinputlimit` is zero which means no limit.
```

Comment:
"is zero which means" --> "is zero, which means"

Reply:
ok